src/dotnet/Fable.Tools/State.fs

```diff
@@ -260,9 +261,22 @@ let compile (com: Compiler) (project: Project) (fileName: string) =
             Fable2Babel.Compiler.createFacade fileName lastFile
     else
         printfn "Compile %s" fileName
-        FSharp2Fable.Compiler.transformFile com project project.CheckedProject fileName
-        |> Fable2Babel.Compiler.transformFile com project
-    |> addLogs com
+        let usingFableCore =  (com :> ICompiler).Options.fableCore = "fable-core"
+        let nextCompiler = 
+            match usingFableCore with
+            | false -> com
+            | true -> 
+                let prevPlugins = (com :> ICompiler).Plugins
+                let prevOptions = (com :> ICompiler).Options
+                let fableCoreLocation = 
+                    let paketDirectory = project.PaketDirectory
+                    IO.Path.Combine(paketDirectory, "packages", "Fable.Core", "fable-core") 
+                    |> Parser.makePathRelative
+                let nextOptions = { prevOptions with fableCore = fableCoreLocation }
+                Compiler(nextOptions, prevPlugins)
```

Comment:
This is a bit nitpicking, but when a piece code is long it seems to take more visual importance. The `compile` function is small so now it looks as if `fable-core` resolution is the core of it. I'd try to make the code more compact and add a comment to make the meaning of it clearer to future readers:

```fsharp
        let com =
            // Resolve the fable-core location if not defined by user
            if (com :> ICompiler).Options.fableCore = "fable-core" then
                let fableCoreLocation =
                    IO.Path.Combine(project.PaketDirectory, "packages", "Fable.Core", "fable-core")
                    |> Parser.makePathRelative
                Compiler({ (com :> ICompiler).Options with fableCore = fableCoreLocation }, (com :> ICompiler).Plugins)
            else com
```

> We can also add the `Options` and `Plugins` members directly to the `Compiler` type so it's not necessary to cast it to `ICompiler` all the time.